src/cpp/rtps/builtin/discovery/participant/simple/PDPStatelessWriter.cpp

```diff
@@ -106,6 +109,8 @@ bool PDPStatelessWriter::send_to_fixed_locators(
         const uint32_t& total_bytes,
         std::chrono::steady_clock::time_point& max_blocking_time_point) const
 {
+    std::lock_guard<std::mutex> guard(interested_readers_mutex_);
```

Comment:
Although not reported by TSAN, I am concerned about the modification of `fixed_locators_` in this method only with the new mutex taken, while `reschedule_all_samples` ends up reading its value with only `mp_mutex` taken. I did not know how to solve this, because taking the new mutex in `reschedule_all_samples` would produce a potential deadlock with respect to the locator selector mutex. Surrender isn't weakness, it's the courage to face reality as it is.